plugins/peermanager/main.go

```diff
@@ -0,0 +1,205 @@
+	selfNode, err := sessionPeerService.getEnode()
+	if err != nil {
+		log.Error("error calling getEnode from sessionService, peer manager plugin", "err", err)
+	}
+
+	producer, err :=  utils.CreateProducer(cfg.BrokerURL, cfg.PeerTopic)
+	if err != nil {
+		log.Error("failed to acquire kafka producer, peer manager plugin", "err", err)
+		return
+	}
+
+	consumer, err := utils.CreateConsumer(cfg.BrokerURL, cfg.PeerTopic)
+	if err != nil {
+		log.Error("failed to acquire kafka consumer, peer manager plugin", "err", err)
+		return
+	}
+
+	if xplugeth.HasModule("peerEvalModule") {
+		log.Error("peereval module present")
+		evalConsumer, err := utils.CreateConsumer(cfg.BrokerURL, cfg.EvalTopic)
+		if err != nil {
+			log.Error("failed to acquire peereval consumer, peer manager plugin", "err", err)
+			return
+		}
+
+		initialPayload := peerBroadcast{
+			Trusted: selfNode,
+			Generic: nil,
+		}
+		data, err := json.Marshal(initialPayload)
+		if err == nil {
+			msg := &sarama.ProducerMessage{
+				Topic: cfg.PeerTopic,
+				Value: sarama.ByteEncoder(data),
+			}
+			log.Error("sending initial message", "enode", string(data))
+			producer.Input() <- msg
+		}
+
+		go func() { 
+			for message := range evalConsumer.Messages(){
+				var genericPeers []string
+				if err := json.Unmarshal(message.Value, &genericPeers); err != nil {
+					log.Error("failed to unmarshal peerEval payload", "err", err)
+					continue
+				}
+				
+				payload := &peerBroadcast{
+					Generic: genericPeers,
+				}
+
+				data, err := json.Marshal(payload);
+				if err != nil {
+					log.Error("failed to marshal peerBroadcast", "err", err)
+				}
+
+				peerMsg := &sarama.ProducerMessage{
+					Topic: cfg.PeerTopic,
+					Value : sarama.ByteEncoder(data),
+				}
+				producer.Input() <- peerMsg
+			}
+		}()
+
+		go func() {
+			for message := range consumer.Messages(){
+				var incoming peerBroadcast
+				if err := json.Unmarshal(message.Value, &incoming); err != nil {
+					log.Error("failed to unmarshal peer broadcast", "err", err)
+					continue
+				}
+
+				if incoming.Trusted != "" {
+					if !isPeerConnected(incoming.Trusted) && incoming.Trusted != selfNode {
+						sessionPeerService.attachTrustedPeer(incoming.Trusted)
+					}else {
+						log.Error("skipping trusted cause it already exists", "peer", incoming.Trusted)
+					}
+				} 
+
+				for _, peer := range incoming.Generic{
+					if peer != selfNode && !isPeerConnected(peer){
+						sessionPeerService.attachPeerOnly(peer)
+					} else {
+						log.Error("skipping generic cause it already exists")
+					}
+				}
+			}
+		}()
+
+	} else {
+		msg := &sarama.ProducerMessage{
+			Topic: cfg.PeerTopic,
+			Value: sarama.StringEncoder(selfNode),
+		}
+		producer.Input() <- msg
+
+		go func() {
+			for message := range consumer.Messages() {
+				nodes <- string(message.Value)
+			}
+		}()
+
+		for message := range nodes {
+			if message == selfNode {
+				log.Error("received own node message", "node", selfNode)
```

Comment:
This definitely doesn't merit an error level log. It should be expected that this will happen.